src/components/Onboarding/index.js

```diff
@@ -7,139 +7,137 @@ import { connect } from 'react-redux'
 import styled from 'styled-components'
 
 import type { T } from 'types/common'
+import type { OnboardingState } from 'reducers/onboarding'
 
 import { saveSettings } from 'actions/settings'
+import { nextStep, prevStep, jumpStep } from 'reducers/onboarding'
+import { getCurrentDevice } from 'reducers/devices'
+
+// TODO: re-write it without auto lock, fixed width of the password modal, not dynamic titles
+import { unlock } from 'reducers/application'
 
 import Box from 'components/base/Box'
-import Button from 'components/base/Button'
-
-const STEPS = [
-  {
-    title: ({ t }) => t('onboarding:step1.title'),
-    render: ({ t }: StepProps) => <Box>{t('onboarding:step1.greetings')}</Box>,
-  },
-  {
-    title: ({ t }) => t('onboarding:step2.title'),
-    render: ({ t }: StepProps) => <Box>{t('onboarding:step2.greetings')}</Box>,
-  },
-  {
-    title: ({ t }) => t('onboarding:step3.title'),
-    render: ({ t }: StepProps) => (
-      <Box>
-        {t('onboarding:step3.greetings')}
-        <Box bg="grey" align="center" justify="center" style={{ width: 200, height: 200 }}>
-          {'step 3 image'}
-        </Box>
-        {t('onboarding:step3.description')}
-      </Box>
-    ),
-  },
-  {
-    title: ({ t }) => t('onboarding:step4.title'),
-    render: ({ t }: StepProps) => <Box>{t('onboarding:step4.greetings')}</Box>,
-  },
-]
+
+import Start from './steps/Start'
+import InitStep from './steps/Init'
+import OnboardingBreadcrumb from './OnboardingBreadcrumb'
+import ChooseDevice from './steps/ChooseDevice'
+import ChoosePIN from './steps/ChoosePIN'
+import WriteSeed from './steps/WriteSeed'
+import GenuineCheck from './steps/GenuineCheck'
+import SetPassword from './steps/SetPassword'
+import Analytics from './steps/Analytics'
+import Finish from './steps/Finish'
+
+const STEPS = {
+  init: InitStep,
+  chooseDevice: ChooseDevice,
+  choosePIN: ChoosePIN,
+  writeSeed: WriteSeed,
+  genuineCheck: GenuineCheck,
+  setPassword: SetPassword,
+  analytics: Analytics,
+  finish: Finish,
+  start: Start,
+}
 
 const mapStateToProps = state => ({
   hasCompletedOnboarding: state.settings.hasCompletedOnboarding,
+  onboarding: state.onboarding,
+  getCurrentDevice: getCurrentDevice(state),
 })
 
 const mapDispatchToProps = {
   saveSettings,
+  nextStep,
+  prevStep,
+  jumpStep,
+  unlock,
 }
 
 type Props = {
   t: T,
   hasCompletedOnboarding: boolean,
   saveSettings: Function,
+  onboarding: OnboardingState,
+  prevStep: Function,
```

Comment:
we need to stop using `Function` in flow. either use `() => void` or define the proper type, for better type quality